llvm/lib/Target/AIE/AIE2RegisterInfo.cpp

```diff
@@ -457,3 +457,32 @@ AIE2RegisterInfo::getCoveringSubRegs(const TargetRegisterClass &RC) const {
   }
   return Subregs;
 }
+bool AIE2RegisterInfo::isVecOrAccRegClass(const TargetRegisterClass *RC) const {
+  // ******** Vector classes ********
+  if (RC->getID() == AIE2::VEC256RegClassID ||
+      find(VEC256Superclasses, RC) != std::end(VEC256Superclasses))
+    return true;
```

Comment:
If you look in the function just above, we use things like `AIE2::VEC1024RegClass.hasSubClassEq(&RC)`, which I think is a bit more readable. Could you maybe rewrite those ifs using `hasSubClassEq()`?

Reply:
yeah you are right, i got confused with subclass in the sense of "C++ inherited class" and AIE subclass (X4 is a subclass/register of Y2)